spec/index.bs

```diff
@@ -610,36 +614,32 @@ dictionary GPUObjectDescriptorBase {
 
 ## Invalid Internal Objects &amp; Contagious Invalidity ## {#invalidity}
 
-If an object is successfully created, it is <dfn dfn>valid</dfn> at that moment.
-An [=internal object=] may be <dfn dfn>invalid</dfn>.
-It may become [=invalid=] during its lifetime, but it will never become valid again.
+Object creation operations in WebGPU are internally asynchronous, so they don't fail with exceptions.
+Instead, returned objects may refer to [=internal objects=] which are either
+<dfn dfn>valid</dfn> or <dfn dfn>invalid</dfn>.
+An object may become [=invalid=] during its lifetime, but it will never become [=valid=] again.
 
-Issue: Consider separating "invalid" from "destroyed".
-This would let validity be immutable, and only operations involving devices, buffers, and textures
-(e.g. submit, map) would check those objects' `[[destroyed]]` state (explicitly).
+Objects are invalid at creation if it wasn't possible to create them.
+This can happen, for example, if the [=object descriptor=] doesn't describe a valid
+object, or if there is not enough memory to allocate a resource.
 
-<div class=note>
-    [=Invalid=] objects result from a number of situations, including:
-
-      - If there is an error in the creation of an object, it is immediately invalid.
-        This can happen, for example, if the [=object descriptor=] doesn't describe a valid
-        object, or if there is not enough memory to allocate a resource.
-      - If an object is explicitly destroyed (e.g. {{GPUBuffer/destroy()|GPUBuffer.destroy()}}),
-        it becomes invalid.
-      - If the [=device=] that owns an object is lost, the object becomes invalid.
-</div>
+Most [=internal objects=] won't become [=invalid=] after they are created, but still
+may become unusable, e.g. if the owning device is [=lose the device|lost=] or
+{{GPUDevice/destroy()|destroyed}}, or the object has a special internal state,
+like buffer state [=buffer state/destroyed=].
```

Comment:
I think so because I think it's better to have validity still be immutable for "nestable" objects, e.g. GPUBuffers/GPUTextures can be referenced inside GPUBindGroups or GPUCommandBuffers.

Merging them would have the awkward effect of validating the "valid" state twice: once at bindgroup/commandbuffer creation, and once at submit(). I don't think there's _strictly_ anything wrong with this, but I think it's scary because it _looks_ like redundant double-validation but isn't.